runtime error on recent compiled master

I recently pulled in the latest in master.
I fist got error while compiling:

rtapi/uspace_rtapi_main.cc:49:10: fatal error: sys/capability.h: No such file or directory
   49 | #include <sys/capability.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.

I then  sudo apt-get install libcap-dev to fix that.
It compiled fine but when I run anything I get an error.

error message in terminal:

Machine configuration file is 'qtdragon_inch.ini'
Traceback (most recent call last):
  File "/home/chris/emc/bin/linuxcnc_check_ini", line 25, in <module>
    import linuxcnc
ImportError: /home/chris/emc/lib/python/linuxcnc.so: undefined symbol: _ZN8linuxcnc7IniFile14mapLinearUnitsERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
linuxcnc_check_ini validation failed


I use mint 22.3 (with no problem till now)

Any idea what to do to fix this?

> rtapi/uspace_rtapi_main.cc:49:10: fatal error: sys/capability.h: No such file or directory 49 | #include <sys/capability.h> | ^~~~~~~~~~~~~~~~~~ compilation terminated.
> I then sudo apt-get install libcap-dev to fix that. It compiled fine but when I run anything I get an error.

There is apparently missing a configure test.


> error message in terminal:
> Machine configuration file is 'qtdragon_inch.ini' Traceback (most recent call last): File "/home/chris/emc/bin/linuxcnc_check_ini", line 25, in import linuxcnc ImportError: /home/chris/emc/lib/python/linuxcnc.so: undefined symbol: _ZN8linuxcnc7IniFile14mapLinearUnitsERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE linuxcnc_check_ini validation failed
> Any idea what to do to fix this?

This usually happens when you have stale generated objects or libraries in your tree. The proper action is to clean your tree and build from fresh. Please note that a `make clean` is not sufficient because changed build rules may leave old stuff behind. The normal procedure is:
```sh
# --> Save your local changes <--
cd .../clone_base_dir/src

# Make sure that any /untracked/ files are safely saved if you want to keep them
# The following instruction will remove anything not belonging to the git tree
git clean -dxf ..

./autogen.sh
./configure
make -O -j$(nproc)
```

--- a/src/configure.ac
+++ b/src/configure.ac
@@ -1544,6 +1544,9 @@ if test "x$HAVE_LIBFMT" = "xno"; then
     AC_MSG_ERROR([libfmt header not found. Install with 'sudo apt-get install libfmt-dev'])
 fi
 
+AC_CHECK_HEADERS([sys/capability.h], [], [
+    AC_MSG_ERROR([libcap header not found. Please install libcap-dev])
+])
 AC_CHECK_HEADERS([editline/readline.h histedit.h], [], [
     AC_MSG_ERROR([libedit headers not found. Please install libedit-dev or libedit-devel.])
 ])

